common/app/common/Box.scala

```diff
@@ -0,0 +1,34 @@
+package common
+
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.concurrent.Future
+import scala.util.{Try, Success, Failure}
+
+abstract class Box[T] {
+  def get(): T
+  def apply(): T
+
+  def send(t: T): Unit
+  def send(f: T => T): Unit
+
+  def alter(t: T): Future[T]
+  def alter(t: T => T): Future[T]
+}
+
+object Box {
+  def apply[T](initialValue: T): Box[T] = new AtomicRefBox[T](initialValue)
+}
+
+private class AtomicRefBox[T](initialValue: T) extends Box[T] {
```

Comment:
Note the omission of `modify`, `map` and `flatMap` (https://github.com/guardian/box/blob/master/src/main/scala/com/gu/Box.scala#L38). These functions aren't used in `frontend` and the `modify` function has not been implemented correctly in `box`.